### PR TITLE
Language support for Modelica

### PR DESCRIPTION
**Pull request by dietmarw**
## Purpose
Adding language support and icon art for [Modelica](https://modelica.org).

## Logo Reference

m_ball

### CODE REVIEW

File: languages.yaml [hunk 1]
```diff
@@ -1605,6 +1605,32 @@ Markdown:
...
+Modelica:
+  type: programming
+  ascii: |
+    {0}          #######     ########    #########
+    {0}        ########### ########################
+    {0}      #######  ###########  ########### ####
+    {0}    ######      #######     ########   #####
+    {0}    #####      ######      #######    ##### 
+    {0}    ###      ######       ######     #####  
+    {0}            ######       #####      #####
```
Comment (spenserblack) [inline languages.yaml:1617]: Looks like some of these lines are a bit too wide. [We enforce a 40-character maximum width at compile time](https://github.com/o2sh/onefetch/actions/runs/7636714914/job/20804507063?pr=1262#step:5:632).

**Review by spenserblack — CHANGES_REQUESTED** [review]
Merging this won't be possible right now because our current tool for language detection [doesn't support Modelica](https://github.com/XAMPPRocky/tokei/blob/5f755db185937b33f38a30e7f6facfa5be0f146a/languages.json). We'd need tokei to support Modelica, and also have a new release featuring that support.

That's the most significant blocking issue currently, though we may [eventually switch tools for language detection](https://github.com/o2sh/onefetch/pull/1152).

**Comment by dietmarw** [comment]
I noticed the missing tokei support too and have now submitted a PR for that: https://github.com/XAMPPRocky/tokei/pull/1061

**Comment by dietmarw** [comment]
@spenserblack Regarding #1152 , is there anything I could already do wrt adding Modelica in case that is resolved before 
https://github.com/XAMPPRocky/tokei/pull/1061 ?

**Comment by spenserblack** [comment]
Yeah you can make an issue or PR into https://github.com/spenserblack/gengo if you want. What's most important is knowledge, not code, so an issue is sufficient, but PRs are very welcome!
I can't say when we'll switch over from tokei to gengo, but I'm pretty confident that we can at least deploy new languages more rapidly (and also resolve language ambiguity) when we do.

File: languages.yaml [hunk 2]
```diff
@@ -1605,6 +1605,32 @@ Markdown:
...
+    {0}###      ######       ######     #####  
+    {0}        ######       #####      #####   
+    {0}      #####       ######       ###     
+    {0}      #####       ######                
+    {0}    #####        ####     {1}***          
+    {0}  #####          ##     {1}*******        
+    {0} #####                 {1}*********       
+    {0}####                   {1}*********       
+    {0}##                      {1}*******               
+    {0}                          {1}***
```
Comment (spenserblack) [inline languages.yaml:1625]:
```suggestion
    {0}      #######     ########    #########
    {0}    ########### ########################
    {0}  #######  ###########  ########### ####
    {0}######      #######     ########   #####
    {0}#####      ######      #######    #####
    {0}###      ######       ######     #####
    {0}        ######       #####      #####
    {0}      #####       ######       ###
    {0}      #####       ######
    {0}    #####        ####     {1}***
    {0}  #####          ##     {1}*******
    {0} #####                 {1}*********
    {0}####                   {1}*********
    {0}##                      {1}*******
    {0}                          {1}***
```

**Review by spenserblack — APPROVED** [review]
Alright, contribution LGTM. The only thing that's left is us resolving upstream issues, so I'm approving this.

**Comment by dietmarw** [comment]
Upstream has just merged the addition: https://github.com/XAMPPRocky/tokei/pull/1061

**Comment by spenserblack** [comment]
We'll need to wait for the next release.

**Review by o2sh — APPROVED** [review]
Thanks a lot for your contribution @dietmarw 🎉 

I took the liberty to clean the ASCII logo slightly https://vercel-gjf16v1xk-o2shs-projects.vercel.app/#Modelica

The PR should be ready to be merged

**Comment by dietmarw** [comment]
> Thanks a lot for your contribution @dietmarw 🎉
> 
> I took the liberty to clean the ASCII logo slightly https://vercel-gjf16v1xk-o2shs-projects.vercel.app/#Modelica
> 
> The PR should be ready to be merged

Thanks @o2sh :-)

**Pull request merged**